sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala

```diff
@@ -690,6 +697,9 @@ class TungstenAggregationIterator(
         val mapMemory = hashMap.getPeakMemoryUsedBytes
         val sorterMemory = Option(externalSorter).map(_.getPeakMemoryUsedBytes).getOrElse(0L)
         val peakMemory = Math.max(mapMemory, sorterMemory)
+        totalPeakMemory += peakMemory
+        maxPeakMemory += peakMemory
+        totalSpilledSize += TaskContext.get().taskMetrics().memoryBytesSpilled - spilledSizeBefore
```

Comment:
this seems fine, but I wonder what happens if you have multiple threads per task, i.e. if another thread spills a lot of things in the mean time in a different operator. Is that a potential problem?